pkg/controllers/management/multiclusterapp/handler_multiclusterapp.go

```diff
@@ -130,3 +150,97 @@ func (r *MCAppRevisionController) sync(key string, mcappRevision *v3.MultiCluste
 	}
 	return mcappRevision, nil
 }
+
+func (p *ProjectController) sync(key string, project *v3.Project) (runtime.Object, error) {
+	if project != nil && project.DeletionTimestamp == nil {
+		return project, nil
+	}
+	splitKey := strings.SplitN(key, "/", 2)
+	if len(splitKey) != 2 || splitKey[0] == "" || splitKey[1] == "" {
+		return project, fmt.Errorf("invalid project id %s", key)
+	}
+	clusterName, projectName := splitKey[0], splitKey[1]
+	key = fmt.Sprintf("%s:%s", clusterName, projectName)
+	mcApps, err := p.mcAppsLister.List(namespace.GlobalNamespace, labels.NewSelector())
+	if err != nil {
+		return project, err
+	}
+	for _, mcApp := range mcApps {
+		if mcApp.DeletionTimestamp != nil {
+			continue
+		}
+		var toUpdate *v3.MultiClusterApp
+		for i, target := range mcApp.Spec.Targets {
+			if target.ProjectName == key {
+				toUpdate = mcApp.DeepCopy()
+				toUpdate.Spec.Targets = append(toUpdate.Spec.Targets[:i], toUpdate.Spec.Targets[i+1:]...)
+				if err := p.updateAnswers(clusterName, key, toUpdate); err != nil {
+					return project, err
+				}
+				break
+			}
+		}
+		if toUpdate != nil {
+			if _, err := p.mcApps.Update(toUpdate); err != nil {
+				return project, fmt.Errorf("error updating mcapp %s for project %s", mcApp.Name, key)
+			}
+		}
+	}
+	return project, nil
+}
+
+func (p *ProjectController) updateAnswers(clusterName string, projectName string, mcapp *v3.MultiClusterApp) error {
+	deleted, err := p.clusterDeleted(clusterName, mcapp)
```

Comment:
When a cluster gets deleted, the sync for project will get called for each project in that cluster. And so will clusterDeleted and clusterOverrideExists. Will it be better to listen on cluster events too to remove the override separately for when a cluster is deleted?